packages/replay/src/replay.ts

```diff
@@ -148,49 +149,72 @@ export class ReplayContainer implements ReplayContainerInterface {
    */
-  public start(): void {
-    this.setInitialState();
+  public initializeSampling(): void {
+    const { errorSampleRate, sessionSampleRate } = this._options;
 
-    if (!this._loadAndCheckSession()) {
+    // if neither sample rate is > 0, then do nothing
+    if (errorSampleRate <= 0 && sessionSampleRate <= 0) {
+      return;
+    }
+
+    // Otherwise if there is _any_ sample rate set, try to load/create session
+    const isSessionSampled = this._loadAndCheckSession();
+
+    if (!isSessionSampled) {
+      // This should only occur if `errorSampleRate` is 0 and was unsampled for
+      // session-based replay. In this case there is nothing to do.
       return;
     }
 
-    // If there is no session, then something bad has happened - can't continue
     if (!this.session) {
-      this._handleException(new Error('No session found'));
+      // This should not happen, something wrong has occurred
+      this._handleException(new Error('Unable to initialize and create session'));
       return;
     }
 
-    if (!this.session.sampled) {
-      // If session was not sampled, then we do not initialize the integration at all.
+    // Only call start if this session is sampled for session-based replays
+    if (this.session && this.session.sampled === 'session') {
+      this._initializeRecording();
       return;
     }
 
-    // If session is sampled for errors, then we need to set the recordingMode
-    // to 'error', which will configure recording with different options.
+    // If not sampled as session-based, it will always be sampled as
+    // error-based if `errorSampleRate` is > 0
     if (this.session.sampled === 'error') {
+      // If session is sampled for errors, then we need to set the recordingMode
+      // to 'error', which will configure recording with different options.
       this.recordingMode = 'error';
+      this._initializeRecording();
     }
 
-    // setup() is generally called on page load or manually - in both cases we
-    // should treat it as an activity
-    this._updateSessionActivity();
+    // There should be no other cases
+  }
 
-    this.eventBuffer = createEventBuffer({
-      useCompression: this._options.useCompression,
-    });
+  /**
+   * Create and start a replay.
+   *
+   * Creates or loads a session, attaches listeners to varying events (DOM,
+   * _performanceObserver, Recording, Sentry SDK, etc)
+   */
+  public start(): void {
+    // TODO: Should we allow you to call start if there is an existing replay in progress?
 
-    this._addListeners();
+    const previousSessionId = this.session && this.session.id;
 
-    // Need to set as enabled before we start recording, as `record()` can trigger a flush with a new checkout
-    this._isEnabled = true;
+    const { session } = getSession({
+      timeouts: this.timeouts,
+      stickySession: Boolean(this._options.stickySession),
+      currentSession: this.session,
+      sessionSampleRate: 1,
+      errorSampleRate: 0,
+    });
```

Comment:
The new `start()` will now forcibly sample for session-based replay.

TODO: Add a 3rd "buffer" recordingMode